osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -922,6 +928,28 @@ public CarouselRoot(BeatmapCarousel carousel)
                 this.carousel = carousel;
             }
 
+            public override void AddChild(CarouselItem i)
+            {
+                CarouselBeatmapSet set = (CarouselBeatmapSet)i;
+                BeatmapSetsByID[set.BeatmapSet.ID] = set;
```

Comment:
It's a nit but from a correctness angle doing this via `.Add()` is a little safer than via the indexer setter, because `.Add()` will throw if the value already exists.